Ligandr/NameToSMILES.py

```diff
@@ -1,34 +1,55 @@
 """
 NameToSMILES.py
-Set up systems to superpose ligands on different conformations
+Converts a PDB alphanumeric code to a Canonical Isomeric SMILES 
+string with explicit hydrogens at a certain pH
 
-Handles the primary functions
 """
 
+import pypdb
+import openbabel
 
-def canvas(with_attribution=True):
-    """
-    Placeholder function to show example docstring (NumPy format)
-
-    Replace this function and doc string for your own project
 
+def getSMILE(chemName):
+    """
+    Finds the SMILES from a 3 or 4 letter code of a chemical located in the PDB
+    
     Parameters
     ----------
-    with_attribution : bool, Optional, default: True
-        Set whether or not to display who the quote is from
+    chemName : str, required
+        The string representing a chemical in the PDB database
 
     Returns
-    -------
-    quote : str
-        Compiled string including quote and optional attribution
+    --------
+    smiles : str
+        SMILES representation of chemical 
     """
+    chem_detail = pypdb.describe_chemical(chemName)
+    smiles = chem_detail["describeHet"]["ligandInfo"]["ligand"]["smiles"]
+    return smiles
 
-    quote = "The code is but a canvas to our imagination."
-    if with_attribution:
-        quote += "\n\t- Adapted from Henry David Thoreau"
-    return quote
-
+def addH(smile_string, pH=7.0):
+    """
+    Protonates a SMILES string for a given pH
 
-if __name__ == "__main__":
-    # Do something if this file is invoked on its own
-    print(canvas())
+    Parameters
+    ------------
+    smile_string : str, required
+        The SMILES format string of a chemical
+    
+    pH : double, optional default = 7.0
+        The pH at which protonation should occur
+    
+    Returns
+    --------
+    protonated_smiles : str
+        The SMILES string representing the protonated ligand
+    """
+    mol = openbabel.OBMol()
```

Comment:
Do you want to use OpenBabel or RDKit or OpenEye? Any is fine, but be aware that the way that SMILES are canonicalized will differ based on the package.

Reply:
We should use RDKit if possible, OpenEye if necessary.

Let's not use OpenBabel.

Reply:
I can't seem to find a way to assign different protonation states based on pH to molecules outside of openbabel.

Reply:
Are you trying to predict pKas, or just enumerate protonation states?

Reply:
Or do you just need smiles with explicit protons?

Reply:
I think the input would be given a pH, return the protonation state. However, I am not entirely sure what is needed for simulation. Could I just enumerate every protonation state?

Reply:
Ah, ok, I see. I would imagine you would actually need a pKa prediction to do what you'd like. Maybe Epik is appropriate here? I think you'd need an actual model of pKas.

Reply:
Ok, I'll look into that! Right now, just going to focus on reaching minimum functionality and then adapting the code later.

Reply:
Yeah I think for now we'll be fine deciding what the protonation state should be "manually" (i.e. this is obvious for co-factors at least) and inputting the SMILES for the protonated ligand.